File: abc381a/abc381a.hs
{-# LANGUAGE OverloadedStrings #-}
import Text.ParserCombinators.ReadP

main :: IO ()
main = getLine >> solve <$> getLine >>= putStrLn

solve ::String -> String
solve s = case readP_to_S ot s of
            [] -> "No"
            ls -> case (last ls) of
                    (True, "") -> "Yes"
                    _ -> "No"
  where
    ot = do
      os <- many (char '1')
      char '/'
      ts <- many (char '2')
      return $ length os == length ts

File: abc381a/abc381a.ml
open Printf
open Scanf

let solve n s =
  let n2 = n / 2 in
  if String.sub s 0 n2 = String.make n2 '1' &&
       String.sub s (n2 + 1) (n - n2 - 1)= String.make n2 '2' &&
         s.[n2] = '/' then "Yes" else "No"

let () =
  scanf "%d %s " solve |> printf "%s\n"
